src/redux/actions/bridge.ts

```diff
@@ -65,35 +69,79 @@ export const updateSelectedAsset = (asset: TokenData):
 
 // this takes the token data input and writes it to the token list
 // this will also update the selected asset if needed
-export const updateTokenData = (tokenData: TokenData):
+export const updateTokenData = (asset: Asset):
   ThunkAction<Promise<void>, {}, {}, AnyAction> => async (
   dispatch: ThunkDispatch<{}, {}, AnyAction>, getState,
 ): Promise<void> => {
   const state = getState() as RootState;
-  const { tokens, selectedAsset } = state.bridge;
+  const { assets, selectedAsset } = state.bridge;
 
   // update token data in token list
-  const updatedTokens = tokens?.map((token: TokenData) => {
-    if (token.token.address === tokenData?.token.address) {
+  const updatedTokens = assets?.map((_asset: Asset) => {
+    if (asset.address === _asset.address) {
       return {
-        ...token,
-        ...tokenData,
+        // TODO: test this?
+        // ...asset,
+        // ..._asset,
+        ..._asset,
+        ...asset,
       };
```

Comment:
this is basically equivalent to Object.assign({}, _asset, asset), right? looks correct but yeh plz test and then remove this TODO if it all seems fine.